common/transport/amqp/src/amqp.ts

```diff
@@ -145,44 +235,190 @@ export class Amqp {
           putToken: (audience, token, callback) => callback(new errors.NotConnectedError()),
         },
         connecting: {
-          _onEnter: (config, connectCallback) => {
-            let connectError = null;
-            const connectErrorHandler = (err) => {
-              connectError = err;
-            };
-
-            this._amqp.on(_amqpClientError, connectErrorHandler);
-            this._amqp.connect(config.uri, config.policyOverride)
-              .then((result) => {
-                debug('AMQP transport connected.');
-                this._amqp.on(_amqpClientError, amqpErrorHandler);
-                this._amqp.removeListener(_amqpClientError, connectErrorHandler);
-                this._fsm.transition('connected', connectCallback, result);
-                return null;
-              })
-              .catch((err) => {
-                this._amqp.removeListener(_amqpClientError, connectErrorHandler);
-                /*Codes_SRS_NODE_COMMON_AMQP_16_003: [The `connect` method shall call the `done` callback if the connection fails.] */
-                this._fsm.transition('disconnected', connectCallback, connectError || err);
-              });
+          _onEnter: (connectionParameters, connectCallback) => {
+            this._rheaContainer.options.sender_options = {properties: {'com.microsoft:client-version': this._config.userAgentString}, reconnect: false};
+            this._rheaContainer.options.receiver_options = {properties: {'com.microsoft:client-version': this._config.userAgentString}, reconnect: false, autoaccept: autoSettleMode};
+            this._connectionCallback = connectCallback;
+            this._indicatedConnectionError = undefined;
+            this._disconnectionOccurred = false;
+            this._sessionCloseOccurred = false;
+            this._connectionCloseOccurred = false;
+            //
+            // According to the rhea maintainers, one can depend on that fact that no actual network activity
+            // will occur until the nextTick() after the call to connect.  Because of that, one can
+            // put the event handlers on the rhea connection object returned from the connect call and be assured
+            // that the listeners are in place BEFORE any possible events will be emitted on the connection.
+            //
+            this._rheaConnection = this._rheaContainer.connect(connectionParameters);
+            manageConnectionHandlers('on');
```

Comment:
>manageConnectionHandlers('on'); [](start = 12, length = 31)

nit: naming is a little generic - try to find something more descriptive: installConnectionEventHandlers?